### PR TITLE
Implement practice exercise `square-root`

### LINKED ISSUE #2359: Implement new practice exercise: `square-root`
The exercise `square-root` has not been implemented yet for the Java track.
The description of the exercise can be found in the [problem specification repository](https://github.com/exercism/problem-specifications/tree/main/exercises/square-root).

How to implement a new exercise for the Java track is described in detail in [CONTRIBUTING.md](https://github.com/exercism/java/blob/master/CONTRIBUTING.md#adding-a-new-exercise). Please have a look there first before starting working on the exercise.
Also please make sure it is clear that you are currently working on this issue, either by asking to be assigned to it, or by opening an empty PR.

When opening an PR, please reference this issue using any of the [closing keywords](https://help.github.com/en/articles/closing-issues-using-keywords).

If you have had a look at the exercise description and you concluded that the exercise might not be possible to implement in the Java language, please leave a comment and describe the problem.

In case you have any further questions, feel free to ask here.

### PR DESCRIPTION
**Pull request by sanderploegsma**
This implements the practice exercise `square-root`, following [its problem specifications](https://github.com/exercism/problem-specifications/tree/main/exercises/square-root).

Fixes #2359  

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

File: config.json
Comment (sanderploegsma) [inline config.json:1931]: One thing that bugs me is that we only have the concepts `for-loops` and `foreach-loops`, and here students probably need a `while` loop, which technically isn't covered by these concepts.

Comment (ErikSchierboom) [reply config.json:1931]: That's unfortunate. Definitely something to look into later.

File: exercises/build.gradle
Comment (ErikSchierboom) [inline exercises/build.gradle:17]: This change might warrant a separate PR, as it changes something relatively important about the tests.

Comment (sanderploegsma) [reply exercises/build.gradle:17]: Yes you’re right, thanks for pointing it out. I guess it should have been included in #2354 since that is the first time we’ve used JUnit 5, but I guess I overlooked it.

Will open a separate PR for this later today, since it’s necessary to correctly test exercises with JUnit 5 in the CI pipeline.

Comment (sanderploegsma) [reply exercises/build.gradle:17]: Opened #2363 to address this. I'll revert the changes here and we'll wait for that PR to be merged first before merging this PR.

**Comment by sanderploegsma** [comment]
Marking as draft since this has to wait for #2363 to be merged first.

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**
